### ISSUE TITLE
Error thrown during run

### ISSUE THREAD

**Issue by lilac** [rolodato/gitlab-letsencrypt#22]
Updated to the latest version, but encountered an error that outputs something like the following
```
By using Let's Encrypt, you are agreeing to the TOS at https://letsencrypt.org/documents/LE-SA-v1.1.1-August-1-2016.pdf
[letiny-core] setChallenge Error:
StatusCodeError: 400 - {"error":"branch_name is missing"}
    at new StatusCodeError (/Users/ivy/.config/yarn/global/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/Users/ivy/.config/yarn/global/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/Users/ivy/.config/yarn/global/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/Users/ivy/.config/yarn/global/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:194:7)
    at Request.<anonymous> (/Users/ivy/.config/yarn/global/node_modules/request/request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:191:7)
    at IncomingMessage.<anonymous> (/Users/ivy/.config/yarn/global/node_modules/request/request.js:1091:12)
    at Object.onceWrapper (events.js:293:19)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:188:7)
    at endReadableNT (_stream_readable.js:975:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
400 - {"error":"branch_name is missing"}
```

**Comment by rolodato** [comment]
@lilac I'll look at this today. Are you using `--jekyll` and/or `--path`? Also, is your GitLab page configured to serve from `master`?

There are some weird things going on with GitLab's API now. Cron'd CI has started failing yesterday all of a sudden and I have no idea why.

**Comment by rolodato** [comment]
@lilac I just retried a CI build and it worked fine, haven't made any changes since the latest release. Could you try again?

**Comment by lilac** [comment]
Yes, I've tried with/o `--jekyll` and `--path`, the same result. Right, from master branch.

I've tried to debug this script, and simulate an api call to get the repo info manually, and found that the `default_branch` field is returned as normal. So the `repo.default_branch` in `lib.js` should be there.

That seems weird to me. I have no idea about the cause till now. Anyway, thanks for your reply.

**Comment by rolodato** [comment]
No problem! My guess is that sometimes GitLab's API screws up and doesn't return the field. I think I've run into this error myself when testing and it went away on its own.

**Comment by lilac** [comment]
@rolodato Sorry I don't get your point of retrying a ci build. I actually run gitlab-le from cli.

**Comment by lilac** [comment]
I rerun the gitlab-le command just now but it is still the same error.

**Comment by rolodato** [comment]
What I meant to say is that I haven't changed anything, so if something broke it's likely on GitLab's end. I have CI set up on this repo to fetch a staging certificate every hour, you can see the results of it here: https://travis-ci.org/rolodato/gitlab-letsencrypt

Could you debug the code to check what the full contents of `repo` are being returned from GitLab?

**Comment by lilac** [comment]
I found the cause, and it is my fault.
I executed a command like this: `TOKEN=<token> gitlab-le --token $TOKEN`, and didn't realize it does not work. Later I changed it to `TOKEN=<token> bash -c "gitlab-le ....."`. Now it works.
Thanks for your help.

**Comment by rolodato** [comment]
No problem, glad you were able to find it! Seems that we're missing a validation, I'll look into that. Running without a token should abort with an error right away.

**Comment by lilac** [comment]
Yes, that would be great.

**Comment by rolodato** [comment]
@lilac I've actually tried doing `TOKEN=... gitlab-le --token $TOKEN ...` and it fails as expected with `Missing required arguments: token`. Could you try this again and check what `args` is being set to here? https://github.com/rolodato/gitlab-letsencrypt/blob/e60a2bb2ac4cb922a3e6201567cc4cffc719e61f/index.js#L3

Make sure to run this in a new shell or run `unset TOKEN` before.

**Comment by lilac** [comment]
You're right. I just tried it, so can confirm that.
My previous env was made complicated by having a token env var set as the github token.

**Comment by rolodato** [comment]
Cool, thanks for the clarification. I haven't tested what happens when you pass in a non-blank but invalid token though, so maybe there's an issue there as well. For example, GitLab could return different fields depending if your API authentication is valid or not.

**Comment by rolodato** [comment]
@lilac Your issue was being caused by using an invalid GitLab API token. I've released 2.1.2 which tries to detect this and throws an error. Thanks for pointing it out!

**Comment by lilac** [comment]
@rolodato Great! Thanks for quick reaction.
